src/main/java/org/symphonyoss/symphony/messageml/bi/BiFields.java

```diff
@@ -2,87 +2,90 @@
-  CASHTAGS("cashtags", BiEventType.MESSAGEML_MESSAGE_SENT),
-  BULLET_LIST("lists", BiEventType.MESSAGEML_MESSAGE_SENT),
-  CARD("cards", BiEventType.MESSAGEML_MESSAGE_SENT),
-  EXPANDABLE_CARDS_COLLAPSED("expandablecards_collapsed", BiEventType.MESSAGEML_MESSAGE_SENT),
-  EXPANDABLE_CARDS_CROPPED("expandablecards_cropped", BiEventType.MESSAGEML_MESSAGE_SENT),
-  EXPANDABLE_CARDS_EXPANDED("expandablecards_expanded", BiEventType.MESSAGEML_MESSAGE_SENT),
-  HEADER("headers", BiEventType.MESSAGEML_MESSAGE_SENT),
-  IMAGE("images", BiEventType.MESSAGEML_MESSAGE_SENT),
-  IMAGE_DATA("images_data", BiEventType.MESSAGEML_MESSAGE_SENT),
-  IMAGE_URL("images_url", BiEventType.MESSAGEML_MESSAGE_SENT),
-  LINE_BREAK("linebreaks", BiEventType.MESSAGEML_MESSAGE_SENT),
-  INPUT_STEP("input_step", BiEventType.MESSAGEML_ELEMENT_SENT),
-  TITLE("has_title", BiEventType.MESSAGEML_ELEMENT_SENT),
-  DEFAULT("default", BiEventType.MESSAGEML_ELEMENT_SENT),
-  PLACEHOLDER("placeholder", BiEventType.MESSAGEML_ELEMENT_SENT),
-  STYLE_COLOR("style_color", BiEventType.MESSAGEML_ELEMENT_SENT),
-  STYLES_CUSTOM("styles_custom", BiEventType.MESSAGEML_MESSAGE_SENT),
-  STYLES_CLASS_TEMPO("styles_class_tempo", BiEventType.MESSAGEML_MESSAGE_SENT),
-  STYLES_CLASS_OTHER("styles_class_other", BiEventType.MESSAGEML_MESSAGE_SENT),
-  TYPE("type", BiEventType.MESSAGEML_ELEMENT_SENT),
-  TYPE_MASKED_TRUE("masked", BiEventType.MESSAGEML_ELEMENT_SENT),
-  TYPE_MASKED_FALSE("normal", BiEventType.MESSAGEML_ELEMENT_SENT),
-  ENTITY_TYPE("entity_type", BiEventType.MESSAGEML_ELEMENT_SENT),
-  MULTI_SUBMIT("is_multi_submit", BiEventType.MESSAGEML_ELEMENT_SENT),
-  LABEL("has_label", BiEventType.MESSAGEML_ELEMENT_SENT),
-  OPTIONS_COUNT("options_count", BiEventType.MESSAGEML_ELEMENT_SENT),
-  VALIDATION("validation", BiEventType.MESSAGEML_ELEMENT_SENT),
-  VALIDATION_MAX("validation_max", BiEventType.MESSAGEML_ELEMENT_SENT),
-  VALIDATION_MIN("validation_min", BiEventType.MESSAGEML_ELEMENT_SENT),
-  VALIDATION_PATTERN("validation_pattern", BiEventType.MESSAGEML_ELEMENT_SENT),
-  VALIDATION_OPTIONS("validation_options", BiEventType.MESSAGEML_ELEMENT_SENT),
-  HIGHLIGHTED_OPTIONS("highlighted_options", BiEventType.MESSAGEML_ELEMENT_SENT),
-  VALIDATION_STRICT("validation_strict", BiEventType.MESSAGEML_ELEMENT_SENT),
-  REQUIRED("required", BiEventType.MESSAGEML_ELEMENT_SENT),
-  MULTI_SELECT("is_multi_select", BiEventType.MESSAGEML_ELEMENT_SENT),
-  MESSAGE_LENGTH("message_length", BiEventType.MESSAGEML_MESSAGE_SENT),
-  ENTITY_JSON_SIZE("entities_json_size", BiEventType.MESSAGEML_MESSAGE_SENT),
-  FREEMARKER("use_freemarker", BiEventType.MESSAGEML_MESSAGE_SENT),
-  POPUPS("popups", BiEventType.MESSAGEML_MESSAGE_SENT),
-  OPENIM("uiactions_openim", BiEventType.MESSAGEML_MESSAGE_SENT),
-  OPENDIALOG("uiactions_opendialog", BiEventType.MESSAGEML_MESSAGE_SENT),
-  COUNT("count", BiEventType.NONE);
+  FORM("form", BiEventType.MESSAGEML_ELEMENT_SENT, null),
```

Comment:
I don't fully understand how do we decide between null and "0" ?

Reply:
Once in the agent:
- I need to be able to differentiate Integers fields from String fields. We have only 2 string fields in the MESSAGEML_ELEMENT_SENT schema: `style_color` and `type` (those are set to null)
- Then we have the element name (`actionmenu`, `textarea`, `timepicker` ...) when generating a payload for an Element event for Button, we don't want to put by default 'timepicker' to 0. So I set them to null to avoid them being added to the default payload.

In a nutshell, we only want to keep the integers and not have the strings at all. This makes for 3 fields that can be string or null, so it should be okay.